encodings/dict/src/compress.rs

```diff
@@ -1,444 +0,0 @@
-use vortex_array::array::{BinaryView, PrimitiveArray, VarBinArray, VarBinViewArray};
-use vortex_array::validity::Validity;
-use vortex_array::variants::PrimitiveArrayTrait;
-use vortex_array::{Array, IntoArray, IntoArrayVariant};
-use vortex_buffer::{BufferMut, ByteBufferMut};
-use vortex_dtype::{match_each_native_ptype, DType, NativePType, PType};
-use vortex_error::{vortex_bail, VortexExpect as _, VortexResult, VortexUnwrap};
-
-use crate::DictArray;
-
-mod private {
-    use vortex_dtype::{half, NativePType};
-
-    /// Value serves as a wrapper type to allow us to implement Hash and Eq on all primitive types.
-    ///
-    /// Rust does not define Hash/Eq for any of the float types due to the presence of
-    /// NaN and +/- 0. We don't care about storing multiple NaNs or zeros in our dictionaries,
-    /// so we define simple bit-wise Hash/Eq for the Value-wrapped versions of these types.
-    #[derive(Debug)]
-    pub struct Value<T>(pub T);
-
-    impl<T> PartialEq<Value<T>> for Value<T>
-    where
-        T: NativePType,
-    {
-        fn eq(&self, other: &Value<T>) -> bool {
-            self.0.is_eq(other.0)
-        }
-    }
-
-    impl<T> Eq for Value<T> where T: NativePType {}
-
-    macro_rules! prim_value {
-        ($typ:ty) => {
-            impl core::hash::Hash for Value<$typ> {
-                fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
-                    self.0.hash(state);
-                }
-            }
-        };
-    }
-
-    macro_rules! float_value {
-        ($typ:ty) => {
-            impl core::hash::Hash for Value<$typ> {
-                fn hash<H: core::hash::Hasher>(&self, state: &mut H) {
-                    self.0.to_bits().hash(state);
-                }
-            }
-        };
-    }
-
-    prim_value!(u8);
-    prim_value!(u16);
-    prim_value!(u32);
-    prim_value!(u64);
-    prim_value!(i8);
-    prim_value!(i16);
-    prim_value!(i32);
-    prim_value!(i64);
-    float_value!(half::f16);
-    float_value!(f32);
-    float_value!(f64);
-}
-
-pub fn dict_encode(array: &Array) -> VortexResult<DictArray> {
-    let dict_builder: &mut dyn DictEncoder = if let Some(pa) = PrimitiveArray::maybe_from(array) {
-        match_each_native_ptype!(pa.ptype(), |$P| {
-            &mut PrimitiveDictBuilder::<$P>::new()
-        })
-    } else if let Some(vbv) = VarBinViewArray::maybe_from(array) {
-        &mut BytesDictBuilder::new(vbv.dtype().clone())
-    } else if let Some(vb) = VarBinArray::maybe_from(array) {
-        &mut BytesDictBuilder::new(vb.dtype().clone())
-    } else {
-        vortex_bail!("Can only encode primitive or varbin/view arrays")
-    };
-    let codes = dict_builder.encode_array(array)?;
-    DictArray::try_new(codes, dict_builder.values())
-}
-
-pub trait DictEncoder {
-    fn encode_array(&mut self, array: &Array) -> VortexResult<Array>;
-
-    fn values(&mut self) -> Array;
```

Comment:
this had another method so you could easily append values to the dictionary so you could use the same logic in dict layout